### DISCUSSION TITLE
How to solve increment correctly?

### DISCUSSION THREAD

**Discussion by Flower-Wang1024** [cvc5/cvc5#11001] [Q&A] [answered]
```python
import cvc5

tm = cvc5.TermManager()
solver = cvc5.Solver(tm)
sm = cvc5.SymbolManager(solver)

solver.setLogic("QF_BV")

parser = cvc5.InputParser(solver, sm)

input1 = "(declare-const x (_ BitVec 4))\n" \
               "(assert (= x #b0001))\n"

parser.setIncrementalStringInput(cvc5.InputLanguage.SMT_LIB_2_6, "myInput")
parser.appendIncrementalStringInput(input1)

# while True:
#     cmd = parser.nextCommand()
#     if cmd.isNull():
#         break
#     cmd.invoke(solver, sm)
# result = solver.checkSat()
# print("Result:", result)

input2 = "(assert (= x #b0101))"
parser.appendIncrementalStringInput(input2)

while True:
    cmd = parser.nextCommand()
    if cmd.isNull():
         break
    cmd.invoke(solver, sm)

result = solver.checkSat()
print("Result:", result)
```
Hello! I'm a student who are learning. I met a problem while trying to solve smt2 incrementally. If I uncomment the code, why can I parse only input1?  Is there something wrong with my code?

**Comment by daniel-larraz** [comment]
Hello,

When running the latest version of the Python API on your original problem, I get:
```
Result: unsat
```
However, if I uncomment the code, I get:
```
Result: sat
Result: sat
```
That seems clearly incorrect. We will take a look at it.

Thank you.

P.S.

To use the latest version of the Python API, you will need to replace:
```
sm = cvc5.SymbolManager(solver)
```
with:
```
sm = cvc5.SymbolManager(tm)
```

We recently introduced this change in the API.

**Reply by Flower-Wang1024** [reply]
ok, thanks a lot! May the version I used is not the latest.

**Comment by daniel-larraz** [comment, marked as answer]
@Flower-Wang1024 The reported issue has been fixed in the main branch (#11070).
